jgrapht-core/src/main/java/org/jgrapht/alg/flow/min_cost/MinimumCostFlowProblem.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * (C) Copyright 2018-2018, by Timofey Chudakov and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * See the CONTRIBUTORS.md file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the
+ * GNU Lesser General Public License v2.1 or later
+ * which is available at
+ * http://www.gnu.org/licenses/old-licenses/lgpl-2.1-standalone.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR LGPL-2.1-or-later
+ */
+package org.jgrapht.alg.flow.min_cost;
+
+import org.jgrapht.Graph;
+import org.jgrapht.alg.interfaces.MinimumCostFlowAlgorithm;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class represents a <a href="https://en.wikipedia.org/wiki/Minimum-cost_flow_problem">
+ * minimum cost flow problem</a>. It serves as input for the minimum cost flow algorithms.
+ *
+ * @param <V> the graph vertex type
+ * @param <E> the graph edge type
+ * @author Timofey Chudakov
+ * @see MinimumCostFlowAlgorithm
+ * @since July 2018
+ */
+public class MinimumCostFlowProblem<V, E> {
+    /**
+     * The flow network
+     */
+    Graph<V, E> graph;
+    /**
+     * The mapping from nodes to the corresponding supplies of the nodes. In every feasible for every node of
+     * the network the sum of all outgoing flows minus the sum of all incoming flows should be equal to the
+     * supply of the node. flow This mapping doesn't need to contain transhipment nodes (with zero supply).
+     */
+    Map<V, Integer> supplyMap;
+    /**
+     * The mapping from edges of the network to the corresponding lower bounds of the flow on edges.
+     * Every feasible flow should satisfy the property that every arc's flow should be no less than
+     * its lower bound. This is an optional part of the problem. This mapping doesn't need to contains
+     * edges with zero lower capacities.
+     */
+    Map<E, Integer> lowerCapacityMap;
+    /**
+     * The mapping from edges of the network to the corresponding upper bounds of the flow on edges.
+     * Every feasible flow should satisfy the property that the flow on an arc doesn't exceeds its
+     * upper bound. This mapping must contain all edges of the network.
+     */
+    Map<E, Integer> upperCapacityMap;
+
+    /**
+     * Constructs a new minimum cost flow problem without arcs' lower bounds.
```

Comment:
I would explicitly specify that the algorithm assumes that the flow costs are assumed to be the weights of the arcs in the graph. Moreover please explicitly mention that these weights can be negative.